docs: Add docs for GCP DHC (#5364)
hashicorp/boundary: website/content/docs/concepts/host-discovery/gcp.mdx

<Tabs>
<Tab heading="CLI">

```shell-session
$ boundary host-catalogs create plugin \
  -scope-id $BOUNDARY_PROJECT_ID \
  -plugin-name gcp \
  -attr disable_credential_rotation=true \
  -attr zone=us-central1-a \
  -attr project_id=env://GCP_PROJECT_ID \
  -attr client_email=env://CLIENT_EMAIL \
  -secret private_key_id=env://PRIVATE_KEY_ID \
  -secret private_key=file://$PRIVATE_KEY_FILE_PATH
```

</Tab>
<Tab heading="Terraform">

  ```hcl
  resource "boundary_host_catalog_plugin" "gcp_host_catalog" {
    name        = "GCP Catalog"
    description = "GCP Host Catalog"
    scope_id    = boundary_scope.project.id
    plugin_name = "gcp"

    attributes_json = jsonencode({
      "zone" = "us-central1-a ",
      "project_id" = var.gcp_project_id,
      "client_email" = var.gcp_client_email,
      "disable_credential_rotation" = true })
    secrets_json = jsonencode({
      "private_key_id" = var.private_key_id,
      "private_key" = var.private_key})
  }
  ```

</Tab>
</Tabs>

### Authentication with service account impersonation
Service account impersonation allows Boundary to authenticate with GCP using a service account that impersonates another service account. 

> This is useful when you want to restrict the permissions of the service account used by Boundary. Two service accounts are required:

Reviewer: ```suggestion
Impersonation is useful when you want to restrict the permissions of the service account Boundary uses. Two service accounts are required:
```

> - `Base service account`: The service account used by Boundary to authenticate with GCP. The service account must have the following roles:

Reviewer: I left suggestions to change passive voice to active voice b/c it is more clear who is performing the action in the sentence.

  -plugin-name gcp \
  -attr disable_credential_rotation=true \
  -attr zone=us-central1-a \
  -attr project_id=env://GCP_PROJECT_ID \
  -attr client_email=env://BASE_SERVICE_ACCOUNT_EMAIL \
  -attr target_service_account_id=env://TARGET_SERVICE_ACCOUNT_EMAIL \
  -secret private_key_id=env://BASE_SERVICE_ACCOUNT_PRIVATE_KEY_ID \
  -secret private_key=file://$BASE_SERVICE_ACCOUNT_PRIVATE_KEY_FILE_PATH
```

</Tab>
<Tab heading="Terraform">

  ```hcl
  resource "boundary_host_catalog_plugin" "gcp_host_catalog" {
    name        = "GCP Catalog"
    description = "GCP Host Catalog"
    scope_id    = boundary_scope.project.id
    plugin_name = "gcp"

    attributes_json = jsonencode({
      "zone" = "us-central1-a ",
      "project_id" = var.gcp_project_id,
      "client_email" = var.gcp_base_service_account_email,
      "target_service_account_id" = var.gcp_target_service_account_email,
      "disable_credential_rotation" = true })
    secrets_json = jsonencode({
      "private_key_id" = var.base_service_account_private_key_id,
      "private_key" = var.base_service_account_private_key})
  }
  ```

</Tab>
</Tabs>

### Authentication with Application Default Credentials (ADC)

> If you are running Boundary on a GCP VM instance, you can use Application Default Credentials (ADC) to authenticate with GCP.

Reviewer: Non-native English speakers struggle with "-ing" words sometimes b/c they can be verbs or nouns in English, so I made suggestions to use present tense verbs instead of present participles.

> Boundary will use the service account associated with the VM instance to authenticate with GCP.

Reviewer: Similar to above, non-native English readers sometimes get confused by future tense b/c they don't know when something is going to happen. So I left suggestions to change future to present tense.

Pass the following fields to Boundary for authentication:

<Tabs>
<Tab heading="CLI">

```shell-session
$ boundary host-catalogs create plugin \
  -scope-id $BOUNDARY_PROJECT_ID \
  -plugin-name gcp \
  -attr disable_credential_rotation=true \
  -attr zone=us-central1-a \
  -attr project_id=env://GCP_PROJECT_ID
```

</Tab>
<Tab heading="Terraform">

  ```hcl
  resource "boundary_host_catalog_plugin" "gcp_host_catalog" {
    name        = "GCP Catalog"
    description = "GCP Host Catalog"
    scope_id    = boundary_scope.project.id
    plugin_name = "gcp"

    attributes_json = jsonencode({
      "zone" = "us-central1-a ",
      "project_id" = var.gcp_project_id,
      "disable_credential_rotation" = true })
  }
  ```

</Tab>
</Tabs>

The `scope-id` and `plugin-name` fields are required when you create a
  dynamic host catalog.

The fields following the `attr` and `secret` flags are specific to GCP and are required by
  Boundary for authentication.

> - `zone` (Required): The deployment area within a region. All host sets in this
>   catalog will be configured for this zone.

Reviewer: ```suggestion
- `zone` (Required): The deployment area within a region. All host sets in this
  catalog are configured for this zone.
```

- `project_id` (Required): The project ID associated with the service account.
- `disable_credential_rotation` (Optional): When set to `true`, Boundary will not rotate the credentials with GCP automatically.
- `client_email` (Optional): The email address associated with the service account. 

>   The email address used to uniquely identify the service account.

Reviewer: ```suggestion
  Boundary uses this email address to uniquely identify the service account.
```

- `target_service_account_id` (Optional): The email address of the service account to impersonate.  

>   This is used for authentication with service account impersonation.

Reviewer: ```suggestion
  Boundary uses this email for authentication with service account impersonation.
```
